test/utils/ErrorHandlerSpec.scala

```diff
@@ -103,9 +103,9 @@ class ErrorHandlerSpec extends UnitSpec with GuiceOneAppPerSuite {
     "return 401 with error body" when {
       "unauthorised and header is supplied" in new Test() {
         val result: Future[Result] = handler.onClientError(requestHeader, UNAUTHORIZED, "test")
-        status(result) shouldBe UNAUTHORIZED
+        status(result) shouldBe ClientOrAgentNotAuthorisedError.httpStatus
```

Comment:
Best be explicit in tests about what you want the behaviour to be. It's not that we want the status to be _whatever_ is in `ClientOrAgentNotAuthorisedError.httpStatus`: we want it to be `UNAUTHORIZED`.
So this line does not need to change as `ErrorHandler` behaviour should remain unchanged. (You will need to fix up `ErrorHandler` accordingly.)